### ISSUE TITLE
Flaky Test: DotNetWatcherTests.RunsWithDotnetWatchEnvVariable

### ISSUE THREAD

**Issue by mikeharder** [aspnet/DotNetTools#285]
```
System.TimeoutException : Task exceeded max running time of 30s at C:\BuildAgent\work\2b0c8ce712895332\.r\DotNetTools\test\Microsoft.DotNet.Watcher.Tools.FunctionalTests\DotNetWatcherTests.cs:26
   at System.Threading.Tasks.TaskExtensions.<OrTimeout>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.DotNet.Watcher.Tools.FunctionalTests.DotNetWatcherTests.<RunsWithDotnetWatchEnvVariable>d__2.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
```

The test appears to be flaky since it usually passes.  However, it's very strange that it failed on 3 separate build configs around the same time:

http://aspnetci/viewLog.html?buildId=206333&tab=buildResultsDiv&buildTypeId=XPlat_Windows_Win7_Universe
http://aspnetci/viewLog.html?buildId=206315&tab=buildResultsDiv&buildTypeId=XPlat_Windows_Win2008r2_Universe
http://aspnetci/viewLog.html?buildId=206333&tab=buildResultsDiv&buildTypeId=XPlat_Windows_Win7_Universe

**Comment by mikeharder** [comment]
Latest failure: http://aspnetci/viewLog.html?buildId=206452&tab=buildResultsDiv&buildTypeId=Lite_UniverseTest

**Comment by natemcmaster** [comment]
The test itself is not flaky. There appears to be problems locating the right version of .NET Core.

The rest of the test output shows this:

```
4/21/2017 11:23:54 AM: post: 'It was not possible to find any compatible framework version'
4/21/2017 11:23:54 AM: post: ''
4/21/2017 11:23:54 AM: post: 'The specified framework 'Microsoft.NETCore.App', version '2.0.0-preview1-002059-00' was not found.'
4/21/2017 11:23:54 AM: post: ''
4/21/2017 11:23:54 AM: post: '  - Check application dependencies and target a framework version installed at:'
4/21/2017 11:23:54 AM: post: ''
4/21/2017 11:23:54 AM: post: '      \'
4/21/2017 11:23:54 AM: post: ''
4/21/2017 11:23:54 AM: post: '  - Alternatively, install the framework version '2.0.0-preview1-002059-00'.'
4/21/2017 11:23:54 AM: post: ''
4/21/2017 11:23:54 AM: recv: 'It was not possible to find any compatible framework version'
4/21/2017 11:23:54 AM: recv: ''
4/21/2017 11:23:54 AM: recv: 'The specified framework 'Microsoft.NETCore.App', version '2.0.0-preview1-002059-00' was not found.'
4/21/2017 11:23:54 AM: recv: ''
4/21/2017 11:23:54 AM: recv: '  - Check application dependencies and target a framework version installed at:'
4/21/2017 11:23:54 AM: recv: ''
4/21/2017 11:23:54 AM: recv: '      \'
4/21/2017 11:23:54 AM: recv: ''
4/21/2017 11:23:54 AM: recv: '  - Alternatively, install the framework version '2.0.0-preview1-002059-00'.'
4/21/2017 11:23:54 AM: recv: ''
```

**Comment by natemcmaster** [comment]
Did we push an updated version of Microsoft.NETCore.App to aspnetcore-ci-dev while this test was running? That might explain the sheared versions. The test started with 2.0.0-preview1-002057-00 but attempted to start a new process with 2.0.0-preview1-002059-00

**Comment by mikeharder** [comment]
Good find, that is exactly what happened.  We should keep the test enabled then.  Larger issue of Pushing packages while builds are running is tracked by https://github.com/aspnet/KoreBuild/issues/228.

**Comment by mikeharder** [comment]
Failed on Travis, and this time I don't see any error about "not possible to find any compatible framework version".

```
Microsoft.DotNet.Watcher.Tools.FunctionalTests.DotNetWatcherTests.RunsWithDotnetWatchEnvVariable [FAIL]
  System.TimeoutException : Task exceeded max running time of 30s at /Users/travis/build/aspnet/DotNetTools/test/Microsoft.DotNet.Watcher.Tools.FunctionalTests/DotNetWatcherTests.cs:26
  Stack Trace:
       at System.Threading.Tasks.TaskExtensions.<OrTimeout>d__1.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at Microsoft.DotNet.Watcher.Tools.FunctionalTests.DotNetWatcherTests.<RunsWithDotnetWatchEnvVariable>d__2.MoveNext()
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
```

https://travis-ci.org/aspnet/DotNetTools/jobs/226563365

**Comment by natemcmaster** [comment]
Good find. I'll take a look.

**Comment by natemcmaster** [comment]
Should be fixed by #289. Closing for now and will reopen if it is still happening on CI.
